### DISCUSSION TITLE
setState() path type safety

### DISCUSSION THREAD

**Discussion by quixot1c** [solidjs/solid#338] [Q&A] [answered]
I am reading the docs before deciding to do a little project with Solid and noticed that if you want to specify the path of a property to update, you do this by using a string or array of strings. Does the typescript compiler know which strings are allowed, especially in nested objects? And does that mean that refactoring a property name should update all the strings referencing that property?

**Comment by quixot1c** [comment, marked as answer]
I decided to try for myself and the answer is kind of. The compiler does know if the strings used match with actual property names, even when you supply multiple in an array, but when refactoring the property name it does not change the string with it. The compiler does start to throw errors now so you do know where to go to fix things.

**Reply by ryansolid** [reply]
Yeah I only compile the JSX. And I make no assumptions about people using TS. The typing for `setState` mostly works, but some people have reported having some edge cases(nullable properties can't be known to exist with normal type narrowing logic flows). It's one of the trickier things to type so always looking at ways to improve.
